### DISCUSSION TITLE
Lgi support on Windows

### DISCUSSION THREAD

**Discussion by denyskon** [xournalpp/xournalpp#4522] [General]
When using Xournal++ on Windows, I'm not able to use GUI plugins because of the missing lua-lgi. Is there any way to install it additionally or to build Xounal++ with lua-lgi included?

**Comment by rolandlo** [comment]
In view of https://github.com/lgi-devs/lgi/issues/98 that might be a difficult task. I haven't tried.

Maybe installing `luarocks` first (see https://github.com/luarocks/luarocks/wiki/Installation-instructions-for-Windows) and then installing `lgi` via luarocks 
```term
luarocks install lgi
``` 
works (more likely it doesn't). In any case I would be interested to know the answer, too.

**Reply by rolandlo** [reply]
**Update**: See simplified installation in [this comment](https://github.com/xournalpp/xournalpp/discussions/4522#discussioncomment-11608162)


I have tried to install `lgi` on Windows and succeeded. Here are the steps (hope I don't forget anything):

1. Install MSYS2 in the default path and at least toolchain, make, git, gtk3, lua as described in the [wiki](https://github.com/xournalpp/xournalpp/blob/master/readme/WindowsBuild.md)
3. Also install gobject-introspection and luarocks via
```term
pacman -S mingw-w64-x86_64-gobject-introspection mingw-w64-x86_64-lua-luarocks
```
4. Clone the `lgi` git repository via:
```term
git clone https://github.com/lgi-devs/lgi.git
```
5. Install the lgi rock via
```
luarocks config --scope system lua_version 5.4
luarocks --local make
```
You may need to create some paths manually (using `mkdir -p <pathname>` on the way) if you don't have at least luarocks 3.9.2.
7. Run `lua` and enter
```term
package.path = "C:\\msys64\\home\\<username>\\.luarocks\\share\\lua\\5.4\\?.lua;" .. package.path
package.cpath = "C:\\msys64\\home\\<username>\\.luarocks\\lib\\lua\\5.4\\?.dll;" .. package.cpath
lgi = require "lgi"
```
and you can now use `lgi`. Similarly for plugins. Instead you can also simply set system environment variables in Windows
`LUA_PATH` with value 
```term
C:\msys64\home\<username>\.luarocks\share\lua\5.4\?.lua;;
```
`LUA_CPATH` with value 
```term
C:\msys64\home\<username>\.luarocks\lib\lua\5.4\?.dll;;
```

Here is a screenshot of a plugin using lgi on Windows: 
![grafik](https://github.com/xournalpp/xournalpp/assets/40485433/39fc1e9c-1774-418d-8ba5-d9f7c8911fa5)

For that to work I had to replace the path separator like in https://github.com/xournalpp/xournalpp/pull/5564

**Edit(s)**: 
- Added missing escape characters in paths.
- Added screenshot
- Added how to use `LUA_PATH` and `LUA_CPATH`
- Fixed a typo
- Added toolchain to dependencies and removed modifying the rockspec (not needed any more)
- Fixed typo in luarocks package

**Comment by jereyes4** [comment]
`mingw-w64-x86_64-gobject-introspection` has the dependency ` mingw-w64-x86_64-gobject-introspection-runtime`, which was broken for me in its current version 1.80.1. If it helps, it could be fixed by replacing `libgirepository-1.0-1.dll` with its version found in 1.78.1, though I doubt that's a good solution.

**Comment by rolandlo** [comment]
**Update**: In the [nightly version](https://github.com/xournalpp/xournalpp/releases/tag/nightly) Xournal++ has `lua-lgi` preinstalled and working out of the box on Windows. 

For non-Xournal++ users wanting to install `lua-lgi`: MSYS2 has updated its [lua-lgi package](https://packages.msys2.org/base/mingw-w64-lua-lgi) in [this commit](https://github.com/msys2/MINGW-packages/commit/2486ecb6c47599485e65581b4d98a3cf89f98467) and the installation is now much simpler than suggested [above](https://github.com/xournalpp/xournalpp/discussions/4522#discussioncomment-8789465): just install [MSYS2](https://www.msys2.org/) and do 
```term
pacman -S mingw-w64-x86_64-lua-lgi
``` 
in the Mingw64 console. 

Compare #6158 for how it was packaged for Xournal++.
